docs/stdlib/constraints.rst

```diff
@@ -9,19 +9,19 @@ Constraints
 
 .. eql:constraint:: std::expression on (expr)
 
-    Arbitrary constraint expression.
+    Represents an arbitrary constraint expression.
 
-    Example of using an ``expression`` constraint to create a custom
-    scalar:
+    The ``expression`` constraint may be used to represent a custom scalar
```

Comment:
I don't think this is accurate. I believe what's happening in the example is that we're _creating_ a custom scalar type rather than _representing_ one.

I like your other changes though. I might also change "may be used" to "may be used as in this example" to drive home the relationship of the sentence to the code sample.

```suggestion
    The ``expression`` constraint may be used as in this example to create a
    custom scalar type:
```